storages/backends/s3boto3.py

```diff
@@ -580,3 +581,9 @@ def get_available_name(self, name, max_length=None):
             name = self._clean_name(name)
             return name
         return super(S3Boto3Storage, self).get_available_name(name, max_length)
+
+    def update_parameters(self, parameters, obj=None, content=None):
+        """
+        Method that can be used to adjust file parameters.
+        """
+        return parameters
```

Comment:
Probably we can move parameters update functionality from _save() into this method

```
    def update_parameters(self, obj=None, content=None, content_type=None):
         """
         Method that can be used to adjust file parameters.
         """
        parameters = self.object_parameters.copy()

         # setting the content_type in the key object is not enough.
         parameters.update({'ContentType': content_type})
 
         if self.gzip and content_type in self.gzip_content_types:
             parameters.update({'ContentEncoding': 'gzip'})
         elif encoding:
             # If the content already has a particular encoding, set it
             parameters.update({'ContentEncoding': encoding})

         return parameters
```